docs/Bundle-Lifecycle.md

```diff
@@ -0,0 +1,48 @@
+## Lifecycle of a Bundle
+
+<img src='./images/codalab-bundle-states.png' />
+
+<br />
+
+There are three different bundle types: `run` bundles, `uploaded` bundles and `make`
+bundles. Each bundle type will enter various states while it is being processed.
+Below are the various bundle state definitions by bundle type:
+
+<br />
+
+### Run Bundles
+
+| State | State Description |
+| - | - |
+| **created** | Bundle has been created but its contents have not been populated yet. |
+| **staged** | Bundle’s dependencies are all ready. Waiting for the bundle to be assigned to a worker to be run. |
+| **starting** | Bundle has been assigned to a worker and waiting for worker to start the bundle. |
+| **preparing** | Waiting for worker to download bundle dependencies and Docker image to run the bundle. |
+| **running** | Bundle command is being executed in a Docker container. Results are uploading. |
```

Comment:
This is a confusing message...could say "Bundle command is being executed in a Docker container or results are being uploaded."  In hindsight, we should have made this two separate states...